### PR TITLE
feat: Added new option for user in the verify callback.

### PR DESCRIPTION
**Pull request by leafaar**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,9 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+## 1.2.2 - 2022-12-21
```
Comment (jnv) [inline CHANGELOG.md:7]: issue: New version entry will be created automatically by our release process. Could you please move your changelong entry into the Unreleased section?

**Review by jnv — CHANGES_REQUESTED** [review]
Thanks for the contribution I have a minor issue with changelog.

But I'm curious. @janhalama Perhaps we could extend ProfileWithMetadata from Express.User? I'm not sure how these Express interfaces work.

**Comment by janhalama** [comment]
> Thanks for the contribution I have a minor issue with changelog.
> 
> But I'm curious. @janhalama Perhaps we could extend ProfileWithMetadata from Express.User? I'm not sure how these Express interfaces work.

This is definitely a bug in definition of verify function callback. The verify function has profile parameter which is in case of our strategy of type `ProfileWithMetadata`. The  parameter `user` in `done` callback is a resolved user which will be assigned by the Strategy to the request context.

`Express.User` is defined as empty object and any object can be assigned to it.

I would either remove `ProfileWithMetaData` from the done user parameter type definition.

```
export class Strategy extends OAuth2Strategy {
...
  constructor(
      options: StrategyOptions,
      verify: (
        accessToken: string,
        refreshToken: string,
        profile: ProfileWithMetaData,
        done: (error: Error | null, user?: Express.User) => void
      ) => void
    ) {
  ...
  }
}
```

OR

use TypeScript Generics: 

```
export class Strategy<User extends Express.User> extends OAuth2Strategy {
...
  constructor(
      options: StrategyOptions,
      verify: (
        accessToken: string,
        refreshToken: string,
        profile: ProfileWithMetaData,
        done: (error: Error | null, user?: User) => void
      ) => void
    ) {
  ...
  }
}
```

**Comment by jnv** [comment]
I'm not sure if @leafaar has a particular use in mind, but I think the first option, ie. pass just `Express.User`, would be sufficient for starters?

**Comment by leafaar** [comment]
I think it's all good just passing the `Express.User`. Thank you!

File: CHANGELOG.md (hunk 1 above)
Comment (leafaar) [reply CHANGELOG.md:7]: Yes, of course!

**Comment by jnv** [comment]
I've opened #28 on top of your PR to keep my changes separate. I will close this PR and we will merge your commits with that other PR. Thanks for pointing out this issue!

**Comment by jnv** [comment]
@leafaar I've just released v1.2.2 with your fix. Thanks again!

**Comment by leafaar** [comment]
Thank you!

**Pull request closed without merging**